src/plugins/dataflow/nodes/controls/dropdown-list-control.tsx

```diff
@@ -172,7 +175,13 @@ export class DropdownListControl extends Rete.Control {
   };
 
   public getSelectionId = () => {
-    return this.props.optionArray.find((opt: any) => optionValue(opt) === this.props.value).id;
+    const optionArray = this.props.optionArray;
+    const value = this.props.value;
+    if (optionArray && value){
+      const selectedOption = optionArray.find((option: any) => optionValue(option) === value);
+      const selectionId = selectedOption ? selectedOption.id : undefined;
+      return selectionId;
+    }
```

Comment:
I'm not sure why - but the original, functionality identical but more terse version of this `getSelectionId` was raising "TypeErrors" during runtime (So - not a type error in TypeScript, but something in the execution called a TypeError?).  Writing it with more procedural checks eliminated that.

Reply:
I wonder if it was because there was no check like on line 182 of the new function? That is, `.id` was being called even if a matching option wasn't found.